app/controllers/application_controller.rb

```diff
@@ -1276,6 +1276,32 @@ def perpage_key(dbname)
     end
   end
 
+  def skip_gtl_columns(view)
+    if defined?(self.class::RESTRICTED_GTL_COLUMNS)
+      self.class::RESTRICTED_GTL_COLUMNS.each do |column, allowed_to_display_method|
+        index_to_delete = try(allowed_to_display_method) ? nil : view.col_order.index(column)
+        if index_to_delete
+          view.cols -= [column] if view.cols
+          view.headers.delete_at(index_to_delete)
+          view.col_order.delete_at(index_to_delete)
+        end
+      end
+    end
+
+    view
+  end
+
+  def load_view_from_db(klass, association, view_suffix, refresh_view, report_name = nil)
```

Comment:
I like the `skip_gtl_columns` approach :+1: 

But let's not add a third way of loading GTL reports please. Two is already more than enough :)

Reply:
(I was thinking calling it from https://github.com/ManageIQ/manageiq-ui-classic/pull/5371/files#diff-38fde2e3fdd09ed7a61246f63ac72d86R39, but YMMV)